testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/kerberos/AbstractKerberosTest.java

```diff
@@ -209,6 +209,36 @@ protected AccessToken assertSuccessfulSpnegoLogin(String clientId, String loginU
         return token;
     }
 
+    protected OAuthClient.AccessTokenResponse assertSpnegoLoginTokenResponse(String loginUsername, String expectedUsername, String password) throws Exception {
+        return assertSpnegoLoginTokenResponse("kerberos-app", loginUsername, expectedUsername, password);
+    }
+
+    protected OAuthClient.AccessTokenResponse assertSpnegoLoginTokenResponse(String clientId, String loginUsername, String expectedUsername, String password) throws Exception {
```

Comment:
IS it possible to update existing method `assertSuccessfulSpnegoLogin` to use your newly added method to avoid so much code duplications?

I am thinking about something around:
```
 protected AccessToken assertSuccessfulSpnegoLogin(String clientId, String loginUsername, String expectedUsername, String password) throws Exception {
        OAuthClient.AccessTokenResponse tokenResponse = assertSpnegoLoginTokenResponse(.. arguments of your new method here ...;

        AccessToken token = oauth.verifyToken(tokenResponse.getAccessToken());
        Assert.assertEquals(userId, token.getSubject());
        Assert.assertEquals(expectedUsername, token.getPreferredUsername());
         return token;
}
```

Reply:
I got rid of that method entirely. The return value AccessToken is only used in one other test, which I adjusted by adding an addition `oauth.verifyToken(response.getAccessToken())` call there.

This feels much cleaner to me, and with relatively minimal overhead of running an extra token verification in a few cases.